### DISCUSSION TITLE
add machine type/category attribute in XML

### DISCUSSION THREAD

**Discussion by ghost** [mamedev/discussions#48] [Ideas]
After studying the output of -listxml a bit, I saw there is no real way to filter out specific game types: arcade, pinball, gambling, ...

We could add a type or category XML attribute inside the `<machine>` tag, with following values, to have a non-subjective classification:

- arcade: all machines of this type, which will represent the original game list of MAME, the arcade machines

- pinball: all machines of this type

- gambling: fruit machine, video slot machine

- gameconsole(*): game console connected to TV, handheld (including old LCD handhelds like Nintendo Game & Watch and tabletop models), any dedicated game console 

- computer(*): all machines of this type

- miscellaneous: synthesizer/musical instrument, calculator, redemption mechnical game, chess board, photo booth, PDA, mobile phone, bowling game, ... Those are ALL machines not inside arcade, pinball, gambling, gameconsole or computer.

Examples:

- arcade:

`<machine name="005" sourcefile="sega/segag80r.cpp" sampleof="005" type="arcade">`

- pinball:

`<machine name="blackblt" sourcefile="pinball/by6803.cpp" ismechanical="yes" sampleof="genpin" type="pinball">`

- gambling: a fruit machine here

`<machine name="v4sixx" sourcefile="barcrest/mpu4vid.cpp" type="gambling">`

- gameconsole:

`<machine name="nes" sourcefile="nintendo/nes.cpp" type="gameconsole">`

- computer:

`<machine name="c64" sourcefile="commodore/c64.cpp" type="computer">`

- miscellaneous: a mobile phone here

`<machine name="noki3210" sourcefile="nokia/nokia_3310.cpp" type="miscellaneous">`

Machine types with (*) will typically contain the machines that support software lists.

What are the advantages of making this?

- Since the list is getting bigger and bigger, I can imagine people only interested in certain type of games would want to filter them out. Currently this is not possible by using the XML. It's easy to add just 1 filtering XML attribute and makes it possible to filter the list that's expanding each release.

- Future MAME frontends/database/website can use it to make filters, if you add a type or category XML attribute inside the `<machine>` tag.

- a frontend that filters on these types, can present you the machines that support software lists (see (*)) in an easy way. You don't need to find them in the huge list (which can be difficult for beginners who don't know the machine names). They also can filter out all arcade, pinball or gambling machines, for people who only want to see those machines.

**Comment by ghost** [comment]
Let me know your thoughts about this!

**Comment by happppp** [comment]
You just keep coming back JJ.
My thoughts? What's the point spending much time discussing anything with you when you'll just ragequit and remove your git account the moment things don't go your way?

**Reply by ghost** [reply]
> You just keep coming back JJ. My thoughts? What's the point spending much time discussing anything with you when you'll just ragequit and remove your git account the moment things don't go your way?

things go my way
i'm now adding the type attribute myself

I DONT NEED YOU FOR THIS

**Comment by galibert** [comment]
In general, we're not interested by adding work for us.

**Reply by ghost** [reply]
> In general, we're not interested by adding work for us.

then don't contribute here if you are not interested

**Comment by ghost** [comment]
OK then feel free to make a TOTAL MESS of the list and keep adding things.
No person on earth will ever know what each machine is or no frontend can sort them

GOOD WORK guys!!!

Why you call me JJ?

**Comment by ghost** [comment]
Who wants to add this?
It's really needed because you don't want a big list without any sorting possible...
